### PR TITLE
[Assist] Add in SSH context Assist endpoints

### PR DESCRIPTION
**Pull request by jakule**
This change introduces web endpoints used by Assist to support SSH context functionality in the WebUI. Now a user will be able to generate Bash command and explain output in the terminal (like application logs).

Posthog events will be introduced in a following PR.

Closes https://github.com/gravitational/teleport.e/issues/1986

### CODE REVIEW

File: lib/ai/model/generationtool.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+func (c *commandGenerationTool) Name() string {
+	return "Command Generation"
+}
+
+func (c *commandGenerationTool) Description() string {
+	// acknowledgement field is used to convince the LLM to return the JSON.
+	// Base on my testing LLM ignores the JSON when the schema has only one field.
+	// Adding additional "pseudo-fields" to the schema makes the LLM return the JSON.
+	return fmt.Sprintf(`Generate a Bash command.
```
Comment (ryanclark) [inline lib/ai/model/generationtool.go:41]: Should we specify something other than bash here? Maybe we could feed the prompt the OS (if we have that available during an SSH session?)

If not, something more generic would avoid the assumption the user's shell is always bash

File: lib/ai/model/generationtool.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+The input must be a JSON object with the following schema:
+%vjson
+{
+	"command": string, \\ The generated command
+	"acknowledgement": boolean \\ Set to true to ackowledge that you understand the formatting
+}
```
Comment (ryanclark) [inline lib/ai/model/generationtool.go:47]: I'm not too familiar with the structure of Assist but could this generate markdown instead? We could stream it to the UI quickly if that was the case, with the suggested command going in between three backticks

Comment (jakule) [reply lib/ai/model/generationtool.go:47]: OpenAI sends the markdown back most of the time.
https://github.com/gravitational/teleport/blob/88959155b46ce0bebb45da4525a8a58c768a8c28/lib/ai/model/prompt.go#L104

**Comment by jakule** [comment]
Friendly ping @ryanclark @xacrimon

**Review by ryanclark — APPROVED** [review]

**Review by xacrimon — APPROVED** [review]

**Comment by xacrimon** [comment]
lgtm but tests seem to be failing

**Comment by jakule** [comment]
> lgtm but tests seem to be failing, both units and flakies

Fixed. That was me fixing too many things at once :unamused:

**Pull request merged**